### PR TITLE
Add `elapsed_years` to `Date` and `DateTime`

### PR DESCRIPTION
**Pull request by yozhgoor**
### Thanks for contributing to chrono!

- [x] Have you added yourself and the change to the [changelog]? (Don't worry
      about adding the PR number)
- [ ] If this pull request fixes a bug, does it add a test that verifies that
      we can't reintroduce it?

Related to #416

I have make this PR for retrieve easily an age from a birthdate
The API is similar to [`std::time::Instant`](https://doc.rust-lang.org/std/time/struct.Instant.html#method.elapsed)

[changelog]: ../CHANGELOG.md

### CODE REVIEW

**Comment by quodlibetor** [comment]
It seems like this would work better as a method on `chrono::Duration`?

**Comment by yozhgoor** [comment]
I need to use a date here, because years are not constant. A year is on average equals to 52.1428571 weeks so this will be really less accurate on `chrono::Duration`.

File: src/lib.rs [hunk 1]
```diff
@@ -594,6 +594,20 @@ mod datetime;
 pub mod format;
 mod round;
 
+/// Constants that can be used by all components.
+pub mod consts {
+    /// Constants of type `f32`.
+    pub mod f32 {
+        /// Number of weeks in a year.
+        pub const WEEK_PER_YEAR: f32 = 52.1775;
```
Comment (cecton) [inline src/lib.rs:602]:
```suggestion
        pub const WEEKS_PER_YEAR: f32 = 52.1775;
```
The comment is correct though :sweat_smile:

File: src/lib.rs [hunk 2]
```diff
@@ -594,6 +594,20 @@ mod datetime;
...
+    }
+    /// Constants of type `f64`.
+    pub mod f64 {
+        /// Number of weeks in a year.
+        pub const WEEK_PER_YEAR: f64 = 52.1775;
```
Comment (cecton) [inline src/lib.rs:607]:
```suggestion
        pub const WEEKS_PER_YEAR: f64 = 52.1775;
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ Versions with only mechanical changes will be omitted from the following list.
 * Add support for microseconds timestamps serde serialization/deserialization (#304)
 * Fix `DurationRound` is not TZ aware (#495)
 * Implement `DurationRound` for `NaiveDateTime`
+* Add a function that calculate the years elapsed between now and a given `Date` or `DateTime`
```
Comment (cecton) [inline CHANGELOG.md:20]:
```suggestion
* Add a function that calculates the number of years elapsed between now and a given `Date` or `DateTime` (#557)
```

**Comment by yozhgoor** [comment]
Thanks @cecton for the spell checking :D

**Comment by yozhgoor** [comment]
I don't know what I do with this PR btw :thinking: 
It's not a problem if this don't fit with the API, at this time we can just close.
I just wanted to solve a problem I had and some others too (see https://github.com/chronotope/chrono/issues/416#issuecomment-687885743).
I don't have the level to trying to calculate the elapsed years on `Duration`. This will make big changes to the API too
This PR is the most accurate i can do for the moment :smile:

**Comment by cecton** [comment]
@Milo123459 this seems ready now :rocket:

**Comment by yozhgoor** [comment]
I fix a conflict in CHANGELOG.md and i think i resolved the issues with `Utc::now` and `Utc::today`.
(I discovered the script to launch the CI locally too)

**Comment by yozhgoor** [comment]
@Milo123459 Can we give it another shot before new changes ?
Thanks

**Comment by yozhgoor** [comment]
So, this seems to be blocked again and I'm not sure why. What needs to be done to move forward? 

This feature cannot be made on Duration because the exact date needs to be known to calculate the number of years exactly. Otherwise it would be an estimation of the numbers of years. My code here gives the exact number of years. 

Do we want this feature or we don't want this feature? I would prefer to get a clear answer if possible. 
Otherwise it feels like I'm being ignored without any reason provided.

File: CHANGELOG.md [hunk 4]
```diff
@@ -744,4 +745,3 @@ and replaced by 0.2.25 very shortly. Duh.)
 ## 0.1.0 (2014-11-20)
 
 The initial version that was available to `crates.io`.
-
```
Comment (Milo123459) [inline CHANGELOG.md:747]: Please remove this useless change

File: src/date.rs [hunk 5]
```diff
@@ -275,6 +276,20 @@ impl<Tz: TimeZone> Date<Tz> {
     pub fn naive_local(&self) -> NaiveDate {
         self.date
     }
+
+    /// Retrieve the elapsed years from now to the given [`Date`].
```
Comment (cecton) [inline src/date.rs:280]: consistency with other's function docs 
```suggestion
    /// Retrieves the elapsed years from now to the given [`Date`].
```

**Comment by djc** [comment]
Unfortunately `std::convert::TryFrom` is not available in our MSRV target, so this needs some work.

**Comment by djc** [comment]
Sorry, can you rebase this again to pull in the fixes from #706?

**Pull request merged**